File: csharp/test/IceBox/configuration/TestI.cs
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

using System.Collections.Generic;

using Test;
using ZeroC.Ice;

public class TestIntf : ITestIntf
{
    private string[] _args;

    public TestIntf(string[] args) => _args = args;

    public string getProperty(string name, Current current) => current.Adapter.Communicator.GetProperty(name) ?? "";

    public IEnumerable<string> getArgs(Current current) => _args;
}

File: java/test/src/main/java/test/Ice/binding/TestI.java
// Copyright (c) ZeroC, Inc.

package test.Ice.binding;

import test.Ice.binding.Test.TestIntf;

public class TestI implements TestIntf {
    TestI() {}

    @Override
    public String getAdapterName(com.zeroc.Ice.Current current) {
        return current.adapter.getName();
    }
}
